src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1688,8 +1688,8 @@ private void WriteRedMinimalFactory(Node nd, bool withStringNames = false)
 
             var hasOptional = minimalFactoryfields.Any(f => !IsRequiredFactoryField(nd, f));
             var hasAttributeOrModifiersList = nd.Fields.Any(f => IsAttributeOrModifiersList(f));
-
-            if (hasOptional && hasAttributeOrModifiersList)
+            var writePragma = hasOptional && (hasAttributeOrModifiersList || nd.Name == "WithElementSyntax");
```

Comment:
I'm lacking context. What is RS0027 and why does `WithElementSyntax` need special treatment here?

Reply:
i marked the place in teh PR where this has impact.  RS2007 says that you should not have APIs that have optional members, if there is an api that has more *non optional* arguments.  Thi sis a suspect, and unmainted roslyn-specific rule that we violate all the time.  Basically, it's fine that one can do `SyntaxFactory.WithElement()` or `SyntaxFactory.WithElement(someWithKeyword, someArgList)`. 

this just suppresses that warning in the generated code (which we do with many other nodes).

Reply:
Thanks
Just to confirm these are the two APIs that would cause that analyzer diagnostic?
```
static Microsoft.CodeAnalysis.CSharp.SyntaxFactory.WithElement(Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentListSyntax? argumentList = null) -> Microsoft.CodeAnalysis.CSharp.Syntax.WithElementSyntax!
static Microsoft.CodeAnalysis.CSharp.SyntaxFactory.WithElement(Microsoft.CodeAnalysis.SyntaxToken withKeyword, Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentListSyntax! argumentList) -> Microsoft.CodeAnalysis.CSharp.Syntax.WithElementSyntax!
```

Reply:
correct.